.github/workflows/tests.yml

```diff
@@ -65,3 +65,38 @@ jobs:
           pip install pytest pytest-describe
       - name: Run Tests
         run: python -m pytest -W ignore::DeprecationWarning
+  e2e:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10"]
+        poetry-version: ["1.3.1"]
+        os: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Upgrade pip
+        run: python -m pip install --upgrade pip
+      - name: Setup Poetry
+        uses: abatilo/actions-poetry@v2
+        with:
+          poetry-version: ${{ matrix.poetry-version }}
+      - name: Install dependencies
+        run: |
+          poetry install --no-interaction --no-root --all-extras -vvv
+          pip install wheel
+          pip install --upgrade setuptools
+          pip install --editable ".[test,ml,medical,dev]"
+          pip install pytest pytest-describe
+      - name: Run Tests
```

Comment:
This could be added as a step under the existing installs from the `tests` job, but I'd say in this instance, I've got a load of pending changes to this area, including this, so I can incorporate it.